packages/components/src/test/tree/controlled/PresentationTreeRenderer.test.tsx

```diff
@@ -331,19 +327,16 @@ describe("PresentationTreeRenderer", () => {
       );
     });
 
-    const result = render(
+    const { getByText, getByRole, user } = render(
       <PresentationTreeRenderer {...baseTreeProps} visibleNodes={visibleNodes} nodeLoader={nodeLoader} onFilterApplied={onFilterAppliedSpy} />,
     );
-
-    const { queryByText, user, getByRole } = result;
-    await waitFor(() => expect(queryByText("A")).to.not.be.null);
-
+    await waitFor(() => getByText("A"));
     // ensure that initially the filter is enabled
     let nodeItem = modelSource.getModel().getNode("A")?.item as PresentationTreeNodeItem;
     expect(nodeItem.filtering?.active).to.not.be.undefined;
 
     const clearFilterButton = await waitFor(() => getByRole("button", { name: "tree.clear-hierarchy-level-filter" }));
-    await user.click(clearFilterButton);
+    await user.pointer({ target: clearFilterButton, keys: "[MouseLeft]" });
```

Comment:
@saskliutas in some tests we click on a button which triggers the tooltip to appear, because the change from the button click would resolve faster than the tooltip (appearing and disappearing) it would cause an act warning as the assertions would pass. Any thoughts how we could avoid this without reducing the click to a pointer or rewriting the tests?

Reply:
We would need to check for both the tooltip appearing and disappearing before we have checks. I guess we could have a function that takes in user (userEvent), component and tooltip text, but I don't know if this workaround would not cause more problems in the future...

Reply:
Does tooltip disappearing also produce `act` warning? I thought it's only during the appearance.

Reply:
Hard to test, but the answer should be yes. Another issue I see with trying to check if the tooltip appeared is that the tooltip would disappear after some delay which can lead to flaky tests.

![image](https://github.com/user-attachments/assets/43408718-acb9-41c8-a468-d03b6e6127e9)

managed to get the DOM (debug), in which you can see the tooltip, but if you try to get all by text the tooltip might already have disapeared.

Reply:
Have you seen this https://floating-ui.com/docs/react#testing? Also I think you could try calling `cleanup` from react-testing-library test body at the end.

Reply:
It seems cleanup fixed the issue, reverted back to using `click`.